src/DurableTask.Netherite/StorageLayer/Faster/StoreWorker.cs

```diff
@@ -281,6 +282,28 @@ internal enum CheckpointTrigger
             Idle
         }
 
+        void LogCheckpointStats()
+        {
+            long inputQueuePositionLag = this.getInputQueuePositionLag();
+
+            this.CheckpointDue(out CheckpointTrigger trigger, out long? compactUntil);
+            var checkpointDueLog = $"Checkpoint statistics: current checkpoint trigger={trigger}, " +
+                $"LastCheckpointedCommitLogPosition={this.lastCheckpointedCommitLogPosition}, " +
+                $"MaxNumberBytesBetweenCheckpoints={this.partition.Settings.MaxNumberBytesBetweenCheckpoints}," +
+                $"CommitLogPosition={this.CommitLogPosition}, " +
+                $"NumberEventsSinceLastCheckpoint={this.numberEventsSinceLastCheckpoint}," +
+                $"MaxNumberEventsBetweenCheckpoints={this.partition.Settings.MaxNumberEventsBetweenCheckpoints}," +
+                $"InputQueuePositionLag={inputQueuePositionLag}," +
+                $"TimeOfNextIdleCheckpoint={this.timeOfNextIdleCheckpoint}";
```

Comment:
could add timeOfFirstRefusedCheckpoint here to help us get a lot of info in just one query.